language/move-vm/runtime/src/loader.rs

```diff
@@ -923,14 +923,18 @@ impl Loader {
     // Verify the kind (constraints) of an instantiation.
     // Both function and script invocation use this function to verify correctness
     // of type arguments provided
-    fn verify_ty_args(&self, constraints: &[AbilitySet], ty_args: &[Type]) -> PartialVMResult<()> {
+    fn verify_ty_args(
+        &self,
+        constraints: &[impl TypeParamGetConstraints],
```

Comment:
same as the other spot `impl IntoIterator<Item = &AbilitySet>` will fix this. Also maybe just `impl IntoIterator<Item = AbilitySet>` since Ability set is just a u8

Reply:
This function in particular accesses the length so the implementation would also need to change. I think the trait is cleaner.

Reply:
Then we can duplicate the function :) Or just copy them out into another vec. Both of these seem better to me than the trait

Reply:
So one thing we could do is leave everything as it was with `&[AbilitySet]` everywhere and add an extra `collect` at the call site when calling from a struct context. I guess you'd be trading extra allocations for the complexity of adding the trait. I particularly like more the trait as I think it's the right abstraction, it captures what's common about function type parameters and struct type parameters, but I'm fine either way.

Reply:
You could also have two different entry points. and then have a shared `_impl` that takes a size and the iterator

Reply:
But I also don't think the extra allocations will be a big deal, like at all. In practice this vector will have like 5 elements at most.